File: SoloLearn/Snowballing Numbers/SnowballingNumbers.rb
def snowball(n)
    total = 0
    for a in 1..n
        val = gets.chomp.to_i
        if val <= total
            return "false"
        end
        total += val
    end
    return "true"
end

n = gets.chomp.to_i
puts snowball(n)

File: SoloLearn/Snowballing Numbers/SnowballingNumbers.cpp
#include <iostream>
using namespace std;

string snowball(int n) {
    int val, total = 0;
    for (int i=0; i<n; i++) {
        cin >> val;
        if (val <= total)
            return "false";
        total += val;
    }
    return "true";
}

int main() {
    int n;
    cin >> n;
    cout << snowball(n);
    return 0;
}